### PR TITLE
Support OpenGL 3

### PR DESCRIPTION
**Pull request by numberZero**
Support OpenGL 3 from minetest/irrlicht#167.

Requires #13280.
Requires minetest/irrlicht#167.

## To do

This PR is Ready for Review
- [x] Merge #13280
- [x] Merge minetest/irrlicht#167
- [x] ~~In this PR, update `misc/irrlichtmt_tag.txt` to refer to the new Irrlicht~~ was updated in master

## How to test

* Build Irrlicht with `USE_SDL2` and `ENABLE_OPENGL3`.
* Copy Irrlicht shaders to `client/shaders/Irrlicht` (as if testing GLES2/desktop)
* Set `video_driver` to `opengl3` in the Minetest config.
* Run Minetest. It should work well, and with no “defaulting to” in the log.

Note: a way to make sure it uses the new driver is *remove* `client/shaders/Irrlicht`. If that breaks rendering, it’s either the proper one or GLES2. It would be better would the *actual* driver used be logged anywhere...

### CODE REVIEW

**Comment by numberZero** [comment]
What’s wrong with win64?
Also, shouldn’t these archives be checksum-verified?

**Comment by numberZero** [comment]
“Rebased.”
The CI will still fail though due to a subtle point I forgot to mention: it’s not enough to *merge* https://github.com/minetest/irrlicht/pull/167, there should also be a way to *build* with it. I.e. a tag on Irrlicht, unless you merge #13215 or an equivalent.

**Comment by numberZero** [comment]
Should work now.

File: src/client/clientlauncher.cpp [hunk 1]
```diff
@@ -211,7 +211,8 @@ bool ClientLauncher::run(GameStartData &start_data, const Settings &cmd_args)
 		m_rendering_engine->get_raw_device()->
 			setWindowCaption((utf8_to_wide(PROJECT_NAME_C) +
 			L" " + utf8_to_wide(g_version_hash) +
-			L" [" + wstrgettext("Main Menu") + L"]").c_str());
+			L" [" + wstrgettext("Main Menu") + L"]" +
+			L" [" + m_rendering_engine->getVideoDriver()->getName() + L"]"	).c_str());
```
Comment (Desour) [inline src/client/clientlauncher.cpp:215]: Nitpick: With opengl3 driver, this doesn't include "OpenGL" for me, i.e. it's just "4.6 (Compatibility Profile) Mesa 23.1.2".

**Comment by Desour** [comment]
>Copy Irrlicht shaders to `client/shaders/Irrlicht` (as if testing GLES2/desktop)

What is the plan here? Do we do it like this until we merge irrlicht into minetest?

**Review by Desour — COMMENTED** [review]
Tested shortly. Works on my machine.
Code also seems reasonable.

File: src/client/clientlauncher.cpp (hunk 1 above)
Comment (numberZero) [reply src/client/clientlauncher.cpp:215]: That’s just what Irrlicht returned.

**Comment by servantoftestator** [comment]
Feel like I'm doing something wrong but having compiled irrlicht-mt-1.9.0mt12 with sdl2, gles2, and opengl3 enabled...

https://pastebin.com/VWdcpQ1V

Running something to replace the compiler recommendations still results in material.setFlag not being found as there is no recommendation for this missing. This is probably user error but I was unable to test this branch.

**Comment by sfan5** [comment]
Git master of MT is (often) only compatible with master of IrrlichtMt.

**Comment by servantoftestator** [comment]
I tried again with latest git of irrlicht-mt and experienced the same errors. This branch is a hundred or so commits behind master anyhow and I'm not squashing and re-basing just to test it when it doesn't compile in the first place in its old environment with older irrlicht.

Commands I ran to fix building somewhat from presumable typos.
find * -type f -print0 | xargs -0 sed -i -e 's/TextureLayer/TextureLayers/g'
find * -type f -print0 | xargs -0 sed -i -e 's/EMF_LIGHTING/EMP_LIGHTING/g'
find * -type f -print0 | xargs -0 sed -i -e 's/EMF_FOG_ENABLE/EMP_FOG_ENABLE/g'
find * -type f -print0 | xargs -0 sed -i -e 's/EMF_BILINEAR_FILTER/EVDF_BILINEAR_FILTER/g'
find * -type f -print0 | xargs -0 sed -i -e 's/EMF_BACK_FACE_CULLING/EMP_BACK_FACE_CULLING/g'
find * -type f -print0 | xargs -0 sed -i -e 's/EMF_ZWRITE_ENABLE/EMP_ZWRITE_ENABLE/g'
find * -type f -print0 | xargs -0 sed -i -e 's/EMF_NORMALIZE_NORMALS/EMP_NORMALIZE_NORMALS/g'

setFlag undefined error https://pastebin.com/DAX7J14J
I don't see setFlag defined in the header for guiScene.cpp and in lib/catch2/catch.hpp it is defined virtual auto setFlag( bool flag ) -> ParserResult = 0;
Including catch headers in guiScene gets me nowhere. Changing everything to use src/settings.h's bool setFlagStr(const std::string &name, u32 flags, gets me nowhere. This branch is very broken for minor issues and I'm unsure how desour and numberzero were able to get it to compile. Again this is probably user error so take it with a grain of salt.

**Comment by numberZero** [comment]
> Feel like I'm doing something wrong but having compiled irrlicht-mt-1.9.0mt12 with sdl2, gles2, and opengl3 enabled...

If you test this PR as-is (without merging with `master`), you need Irrlicht compatible with the MT version this PR branches from. I.e. 1.9.0mt11. `setFlag` was removed in https://github.com/minetest/irrlicht/pull/204, just before the `mt12` release.

**Comment by numberZero** [comment]
Rebased. Without dynamic shadow support this time, sorry.

**Pull request merged**